kong/runloop/handler.lua

```diff
@@ -413,11 +413,8 @@ return {
 
 
       worker_events.register(function(data)
-        -- assume an update doesnt also change the whole entity!
-        if data.operation ~= "update" then
-          log(DEBUG, "[events] Plugin updated, invalidating plugins map")
-          cache:invalidate("plugins_map:version")
-        end
+        log(DEBUG, "[events] Plugin updated, invalidating plugins map")
+        cache:invalidate("plugins_map:version")
```

Comment:
A more efficient approach would be to be able to "query" the plugins_map to determine without the plugin in question is already part of it or not, and rebuild the map only when necessary. We can keep this as a backlog item for now.